sgx_tstd/src/sys/fs.rs

```diff
@@ -195,19 +201,25 @@ impl Iterator for ReadDir {
         }
 
         unsafe {
-            let mut ret = DirEntry { entry: mem::zeroed(), dir: self.clone() };
+            let mut ret = DirEntry {
+                entry: mem::zeroed(),
+                dir: self.clone(),
+            };
             let mut entry_ptr = ptr::null_mut();
             loop {
-                if libc::readdir64_r(self.inner.dirp.0, &mut ret.entry, &mut entry_ptr) != 0 {
-                    if entry_ptr.is_null() {
-                        // We encountered an error (which will be returned in this iteration), but
-                        // we also reached the end of the directory stream. The `end_of_stream`
-                        // flag is enabled to make sure that we return `None` in the next iteration
-                        // (instead of looping forever)
-                        self.end_of_stream = true;
+                match libc::readdir64_r(self.inner.dirp.0, &mut ret.entry, &mut entry_ptr) {
+                    Err(e) => {
+                        if entry_ptr.is_null() {
+                            // We encountered an error (which will be returned in this iteration), but
+                            // we also reached the end of the directory stream. The `end_of_stream`
+                            // flag is enabled to make sure that we return `None` in the next iteration
+                            // (instead of looping forever)
+                            self.end_of_stream = true;
+                        }
+                        return Some(Err(e.into()));
                     }
-                    return Some(Err(Error::last_os_error()));
-                }
+                    _ => {}
+                };
```

Comment:
`match` a Result with an empty arm is equivalent to `if let` which is little more clear 

```rust
if let Err(e) = libc::... {
    if entry_ptr.is_null() {
        self.end_of_stream = true;
    }
    return Some(Err(Error::last_os_error()));
}
```